comp/aggregator/demultiplexer/demultiplexerimpl/demultiplexer_fake_sampler_mock.go

```diff
@@ -22,7 +22,11 @@ import (
 // FakeSamplerMockModule defines the fx options for FakeSamplerMock.
 func FakeSamplerMockModule() fxutil.Module {
 	return fxutil.Component(
-		fx.Provide(newFakeSamplerMock))
+		fx.Provide(newFakeSamplerMock),
+		fx.Provide(func(demux demultiplexerComp.FakeSamplerMock) aggregator.Demultiplexer {
```

Comment:
Can you explain why you need this as it is not common to provide non component type  (except for `Params` type).

Reply:
Since I replaced how a server is started, a new dogstatsd server no longer require a demultiplexer when started anymore. This causes some changes in the test files such as this [function](https://github.com/DataDog/datadog-agent/blob/AMLII-1408-remove-start-method-in-favor-of-on-start-hook-in-dogstatsd-server-component/comp/dogstatsd/server/server_test.go#L417). Or any function that used to require the dogstatsdserver to include a demultiplexer. 

Doing this provide a mock implementation instead of an actual component for those tests.

Reply:
Since I replaced how a server is started, a new dogstatsd server no longer require a demultiplexer when started anymore. This causes some changes in the test files such as this [function](https://github.com/DataDog/datadog-agent/blob/AMLII-1408-remove-start-method-in-favor-of-on-start-hook-in-dogstatsd-server-component/comp/dogstatsd/server/server_test.go#L417). Or any function that used to require the dogstatsdserver to include a demultiplexer. 

Doing this provide a mock implementation instead of an actual component for those tests.